boot/bootchain.go

```diff
@@ -211,8 +224,7 @@ func predictableBootChainsEqualForReseal(pb1, pb2 predictableBootChains) bool {
 	if err != nil {
 		return false
 	}
-	// TODO:UC20: return false if either chains have unasserted kernels
-	return bytes.Equal(pb1JSON, pb2JSON)
+	return bytes.Equal(pb1JSON, pb2JSON) && !pb1.hasUnrevisionedKernels()
```

Comment:
does this need `&& !pb2.hasUnrevisionedKernels()` too ? or is that not an issue because we can't refresh from an asserted kernel to an unasserted kernel ?

Reply:
it's simpler than that, they need to be equal in the first place so the predicate would return the same for either pbc1 and pbc2

Reply:
ah I see yes that makes sense